File: dogpack/apps/1d/advection/smooth_example_interface/plotq1.m
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%
%%   Information you have available to you:
%%
%%     Basic information:
%%              mx:  number of points
%%    [xlow,xhigh]:  min/max values of grid
%%            meqn:  number of equations
%%            maux:  number of aux components
%%           meth1:  spatial order of accuracy
%%
%%   Grid information:
%%              xc: grid points (cell centers), size = (mx,my)
%%
%%   Solution information:
%%           qsoln:  solution sampled on mesh, size = (mx,meqn)
%%             aux:  aux components sampled on mesh, size = (mx,maux)
%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

if time==0
  qex = qsoln;
end

figure(1);
clf;
pt=plot(xc,qex(:,1),'r-');
set(pt,'linewidth',1.5);
hold on;
pz=plot(xc,qsoln(:,1),'bo');
set(pz,'markersize',8)
set(pz,'linewidth',1);
hold off;
axis on; box on; grid off;
axis([0 1 -0.1 1.1]);
set(gca,'plotboxaspectratio',[2 1 1]);
set(gca,'xtick',-2:0.25:2);
set(gca,'ytick',-2:0.5:2);
set(gca,'fontsize',16);
t1 = title(['q(x,t) at t = ',num2str(time),'     [DoGPack]']); 
set(t1,'fontsize',16);

if ( abs(time-round(time)) < 1.0e-10 && time > 0)
  err1 = norm(qsoln-qex,1)/norm(qex,1);
  err2 = norm(qsoln-qex,2)/norm(qex,2);
  disp(' ');
  disp(['   dx = ',num2str(dx,'%0.8e'),['         err1 = ' ...
                          ''],num2str(err1,'%0.8e')]);
  disp(' ');
  disp(['   dx = ',num2str(dx,'%0.8e'),['         err2 = ' ...
                      ''],num2str(err2,'%0.8e')]);
  disp(' ');
end

disp([['Min value of q = ', num2str( min(qsoln(:,1)), '%1.10e' ) ]]);

File: dogpack/apps/1d/advection/smooth_example_interface/plotq1.py
#----------------------------------------------------------
def plotq1(m, space_order, meqn, mx, time, xc, qsoln, auxsoln):
    
    import matplotlib.pyplot as plt
    import numpy as np

    # Exact solution:
    width = 2.0*0.25
    xold = (xc - time)%1.
    qex  = 0.*qsoln[:,0]
    for (i,x) in enumerate(xold):
        if( abs(x-0.5) <= 0.5*width ):
            qex[i] = 1.0;#np.cos(np.pi*(x-0.5)/width)**6

    plt.figure(1)
    plt.clf()
    plt.gca().set_aspect('auto')
    #plt.gca().set_xlim([xc[0],xc[mx-1]])
    #plt.gca().set_ylim([-1.0,2.1])
    plt.plot(xc,qsoln[:,m],'bo')
    #plt.plot(xc,(abs(qsoln[:,m])+1.0e-16), 'bo', linewidth=2.0)
    #tmp1 = "".join(("q(t,x) at t = ",str(time)))
    tmp1 = "L2 Projection"
    title = "".join((tmp1,"     [DoGPack]"))
    plt.title(title)
    plt.tight_layout()
    plt.plot([0.5,0.5],[0.0,2.1],'r')
    print mx/5, max(abs(qex-qsoln[:,m])),time
    #plt.savefig('solnalphais3015discon.pdf')
    plt.pause(0.1)
    plt.draw()
    
###----------------------------------------------------------
